### PR TITLE
fix: Change default transport to fetch

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

...

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

File: src/posthog-featureflags.ts [hunk 1]
```diff
@@ -433,7 +433,7 @@ export class PostHogFeatureFlags {
 
         if (!existing_early_access_features || force_reload) {
             this.instance._send_request({
-                transport: 'XHR',
+                transport: 'fetch',
```
Comment (pauldambra) [inline src/posthog-featureflags.ts:436]: can this just not specify their own transport and use the default instead?

File: src/posthog-surveys.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ export class PostHogSurveys {
                     `/api/surveys/?token=${this.instance.config.token}`
                 ),
                 method: 'GET',
-                transport: 'XHR',
+                transport: 'fetch',
```
Comment (pauldambra) [inline src/posthog-surveys.ts:121]: can this just not specify their own transport and use the default instead?

File: src/web-experiments.ts [hunk 3]
```diff
@@ -151,7 +151,7 @@ export class WebExperiments {
                 `/api/web_experiments/?token=${this.instance.config.token}`
             ),
             method: 'GET',
-            transport: 'XHR',
+            transport: 'fetch',
```
Comment (pauldambra) [inline src/web-experiments.ts:154]: can this just not specify their own transport and use the default instead?

didn't realise there were three when i started commenting on each 🤣

File: src/posthog-featureflags.ts (hunk 1 above)
Comment (benjackwhite) [reply src/posthog-featureflags.ts:436]: Probably.

Comment (pauldambra) [reply src/posthog-featureflags.ts:436]: https://github.com/PostHog/posthog-js/pull/1614

**Pull request merged**